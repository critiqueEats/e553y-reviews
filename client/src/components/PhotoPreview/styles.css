.container {
    box-sizing: border-box;
    display: inline-flex;
    flex-wrap: wrap;
    flex-direction: row;
    width: 348px;
}

.photoCardFull, 
.photoCardHalf {
    border-radius: 10px;
    border: 5px transparent;
    margin: 3px;
    position: relative;
    overflow: hidden;
}

.photoCardFull {
    width: 348px;
    height: 348px;
}

.photoCardHalf {
    width: 168px;
    height: 168px;
}

.thumbnail {
    position: absolute;
    left: 50%;
    top: 50%;
    min-height: 100%;
    min-width: 100%;
    transform: translate(-50%, -50%);
}