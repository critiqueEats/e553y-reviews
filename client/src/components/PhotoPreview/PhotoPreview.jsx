import React from 'react';
import styles from './styles.css';

const PhotoPreview =({photos}) => {
    const photoCount = photos.length;

    if(photoCount === 1) {
        return (
            <div className={styles.container}>
                <div className={styles.photoCardFull}>
                    <img className={styles.thumbnail} src={photos[0]} alt="food picture"/>
                </div>
            </div>
        );
    }
    if(photoCount === 2) {
        return (
            <div className={styles.container}>
                <div className={styles.photoCardHalf}>
                    <img className={styles.thumbnail} src={photos[0]} alt="food picture"/>
                </div>
                <div className={styles.photoCardHalf}>
                    <img className={styles.thumbnail} src={photos[1]} alt="food picture"/>
                </div>
            </div>
        );
    }

    return (
        <div className={styles.container}>
            <div className={styles.photoCardFull}>
                <img className={styles.thumbnail} src={photos[0]} alt="food picture"/>
            </div>
            <div className={styles.photoCardHalf}>
                <img className={styles.thumbnail} src={photos[1]} alt="food picture"/>
            </div>
            <div className={styles.photoCardHalf}>
                <img className={styles.thumbnail} src={photos[2]} alt="food picture"/>
            </div>
        </div>
    )

};

export default PhotoPreview;
