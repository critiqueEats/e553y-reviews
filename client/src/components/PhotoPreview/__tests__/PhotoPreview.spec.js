import React from 'react'
import { shallow, configure, mount } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import jestSerializerHtml from 'jest-serializer-html'

configure({adapter: new Adapter()})

expect.addSnapshotSerializer(jestSerializerHtml)

import PhotoPreview from '../PhotoPreview.jsx'

describe('PhotoPreview Component', function(){
  it('should generate matching html snapshot', function() {
    let testProps = {
      photos:['url1', 'url2', 'url3']
    }

    expect(shallow(<PhotoPreview {...testProps} />).html()).toMatchSnapshot()

  })


  it('should render full width if 1 photo', function() {
    let testProps = {
      photos: ['url1']
    }
    
    let wrapper = shallow(<PhotoPreview {...testProps} />)
    expect(wrapper.find('.photoCardFull')).toHaveLength(1)
  })

  it('should render side by side if 2 photos', function() {
    let testProps = {
      photos: ['url1', 'url2']
    }
    
    let wrapper = shallow(<PhotoPreview {...testProps} />)
    expect(wrapper.find('.photoCardHalf')).toHaveLength(2)
  })
  
  it('should render with correct class if 3 photos', function() {
    
    let testProps = {
      photos: ['url1', 'url2', 'url3']
    }
    
    let wrapper = shallow(<PhotoPreview {...testProps} />)
    expect(wrapper.find('.photoCardHalf')).toHaveLength(2)
    expect(wrapper.find('.photoCardFull')).toHaveLength(1)
  })

  it('should render only first 3 if more than 3 photos', function() {
    
    let testProps = {
      photos: ['url1', 'url2', 'url3', 'url4']
    }
    
    let wrapper = shallow(<PhotoPreview {...testProps} />)

    expect(wrapper.find('.photoCardHalf')).toHaveLength(2)
    expect(wrapper.find('.photoCardFull')).toHaveLength(1)
    
    wrapper.find('img').map(imgWrapper => {

      expect(imgWrapper.prop('src')).not.toBe('url4')

    })
  })
})