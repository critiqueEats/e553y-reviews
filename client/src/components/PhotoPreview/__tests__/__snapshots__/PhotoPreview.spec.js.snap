// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhotoPreview Component should generate matching html snapshot 1`] = `

<div class="container">
  <div class="photoCardFull">
    <img class="thumbnail"
         src="url1"
         alt="food picture"
    >
  </div>
  <div class="photoCardHalf">
    <img class="thumbnail"
         src="url2"
         alt="food picture"
    >
  </div>
  <div class="photoCardHalf">
    <img class="thumbnail"
         src="url3"
         alt="food picture"
    >
  </div>
</div>

`;
